modules/ROOT/pages/dataweave-formats.adoc

```diff
@@ -62,6 +62,75 @@ DataWeave supports these formats (or MIME types) as input and output:
 |===
 // TODO: PLAIN TEXT SECTION? <<format_plain_text>>
 
+[[dw_readers_writers]]
+== DataWeave Readers and Writers
+
+DataWeave can read input data as a whole in-memory, in indexed fashion,
+and for some data formats, part-by-part by streaming the input. When attempting
+to read a large file, it is possible to run out of memory or to impact
+performance negatively. Streaming can improve performance but impacts access to file.
+
+* Indexed and In-Memory: Allow for random access to data because both
+strategies parse the entire document. For these strategies, your DataWeave
+script can access any part of the resulting value at any time.
+** Indexed: Uses indexes over the disk.
+** In-Memory: Parses the entire document in memory.
+
+* Streaming: Allows for sequential access to the file. This strategy partitions
+the input document into smaller items and accesses its data sequentially,
+storing the current item in memory. A DataWeave selector can access the portion
+of the file that is getting read. DataWeave supports streaming for a few
+formats:
+** <<format_csv, CSV>>
+** <<format_json, JSON>>
+
+[[reader_writer_properties]]
+=== Using Reader and Writer Properties
```

Comment:
The Using Reader and Writer Properties section was just moved and converted to an H3.